### PR TITLE
rfc: account management and service integration

### PR DESCRIPTION
**Pull request by nexus49**

### CODE REVIEW

File: rfc/001-account-management-and-service-integration.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+- This document does not detail the process of enabling a service on the portal in general. It presumes that the integrated service is configured to be accessible to all accounts on the portal.
+
+## Options Considered
+
+- An account loosely linked with a service-specific onboarding experience.
+
+### Loosely Coupled Account with a Service-Specific Onboarding Experience
+
+In this scenario, the account management is solely responsible for creating and managing accounts. Services that wish to integrate into the openmfp ecosystem must implement a service-specific onboarding experience. This includes creating a service-specific resources and linking persisting the link to the central account management.
```
Comment (petersutter) [inline rfc/001-account-management-and-service-integration.md]:
```suggestion
In this scenario, the account management is solely responsible for creating and managing accounts. Services that wish to integrate into the openmfp ecosystem must implement a service-specific onboarding experience. This includes creating service-specific resources and persisting the link to the central account management.
```

File: rfc/001-account-management-and-service-integration.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
+# RFC for openMFP Account Management and Service Integration
+
+## Context and Problem Statement
+
+This document provides an outline for the central account management of openmfp and its interaction with contributing services. In this context a account entity is meant to be a grouping entity that groups several resources - similar to a hyperscaler account. In openmfp the account will allow users to group resources of underlying services indirectly. It is also a key entity that other aspects of openmfp like extensions will build upon.
```
Comment (petersutter) [inline rfc/001-account-management-and-service-integration.md]:
```suggestion
This document provides an outline for the central account management of openmfp and its interaction with contributing services. In this context, an account entity is meant to be a grouping entity that groups several resources - similar to a hyperscaler account. In openmfp, the account will allow users to group resources of underlying services indirectly. It is also a key entity that other aspects of openmfp, like extensions, will build upon.
```

**Comment by holgerkoser** [comment]
Overall, the concept of having separate entities (a `Project` resource in the service and an `Account` resource in IAM) provides a clean and clear separation of responsibilities. It seems reasonable to expect the end user to handle the creation of service-specific resources with reference to their current account separately. However, there are some points that need clarification:

1. Is the account information also stored on the `Project` resource (for example, as a label)? This detail is important for understanding how integrated services maintain / retrieve the linked account context. 
   
2. How is a `Project` created at the service CLI level? It appears that IAM rights are required to create projects. How is the account assignment handled in this scenario?

3. How does this concept address the management of local technical users `ServiceAccounts` within the service? Are verifications also carried out through IAM?

4. What is the cardinality of the relationship between `Account` and `Project`? I assume it is `1 - 0..1` or `1 - 0..n`, but can a project also be associated with two accounts?

Clarifying these aspects will help ensure that the system architecture supports both flexibility and robustness in handling integrated services.

**Comment by nexus49** [comment]
> Is the account information also stored on the Project resource (for example, as a label)? This detail is important for understanding how integrated services maintain / retrieve the linked account context.

Yes i think we should store the account on the linked resource, eg. by using a label

> How is a Project created at the service CLI level? It appears that IAM rights are required to create projects. How is the account assignment handled in this scenario?

I think we likely would need a validatingwebhook to check if the use has the necessary permissions to create a project in the context of an account.

> How does this concept address the management of local technical users ServiceAccounts within the service? Are verifications also carried out through IAM?

The concept is agnostic to end-user vs service account actors. From an authorization webhook point for view we could validate access from a technical actor or a user actor. So far this was not yet the focus, but i dont' see a big problem right away.

> What is the cardinality of the relationship between Account and Project? I assume it is 1 - 0..1 or 1 - 0..n, but can a project also be associated with two accounts?

In my opinion I would go forward with the assumption that we have a 0..n cardinality between accounts and projects.

**Pull request merged**